clang/lib/DPCT/ASTTraversal.cpp

```diff
@@ -3215,6 +3215,10 @@ void VectorTypeMemberAccessRule::registerMatcher(MatchFinder &MF) {
                          hasAnyName(SUPPORTEDVECTORTYPENAMES)))))))))))
           .bind("VecMemberExprArrow"),
       this);
+
+  // Want to match "decltype(uint3::x)", but there is no way to judge the
```

Comment:
No inner filter is available for decltypeType(), thus, this matcher will matches all decltypeType. Detail control flow for different types is in runRule().